rsciio/arina/_api.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2025 The HyperSpy developers
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with any project and source this library is coupled.
+# If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import logging
+from pathlib import Path
+
+import h5py
+import hdf5plugin  # noqa F401
+import numpy as np
+
+_logger = logging.getLogger(__name__)
+
+
+def file_reader(
+    filename,
+    lazy=False,
+    navigation_shape=None,
+    rebin_diffraction=1,
+    dtype=None,
+    flatfield=None,
+):
+    """Read arina 4D-STEM datasets.
+
+    Parameters
+    ----------
+    navigation_shape : int, optional
+        Real space dimension of scan. If None, assumes a square acquisition.
```

Comment:
Actually, passing an integer (in addition to the tuple) may be more simple than specifying a tuple with "auto", not sure if "auto" is necessary?

The parameter `navigation_shape` should have the same signature as for other reader and this is also we reuse the docstring:
https://github.com/hyperspy/rosettasciio/blob/be2a6bd4bf26c631fbe564191247b22bb4b0d581/rsciio/mrc/_api.py#L553

https://github.com/hyperspy/rosettasciio/blob/be2a6bd4bf26c631fbe564191247b22bb4b0d581/rsciio/_docstrings.py#L79-L83


However, other reader will need to be updated to support it and it would be better to do it in a follow up PR - ideally before it is released.

Reply:
Yes in this case since the total number of diffraction patterns is known, I tried to leave it flexible. I hope this matches better with the other readers!